src/commands/createFunction/createFunction.ts

```diff
@@ -93,22 +91,29 @@ export async function createFunction(
 
     if (functionAppPath === undefined) {
         const folderPlaceholder: string = localize('azFunc.selectFunctionAppFolderExisting', 'Select the folder containing your function app');
-        functionAppPath = await workspaceUtil.selectWorkspaceFolder(ui, folderPlaceholder);
+        functionAppPath = await workspaceUtil.selectWorkspaceFolder(ext.ui, folderPlaceholder);
     }
 
-    await validateIsFunctionApp(telemetryProperties, outputChannel, functionAppPath, ui);
+    await validateIsFunctionApp(telemetryProperties, functionAppPath);
 
-    const localAppSettings: LocalAppSettings = new LocalAppSettings(ui, tree, functionAppPath);
+    const localAppSettings: LocalAppSettings = new LocalAppSettings(ext.ui, ext.tree, functionAppPath);
 
-    let language: ProjectLanguage = await getProjectLanguage(functionAppPath, ui);
-    let runtime: ProjectRuntime = await getProjectRuntime(language, functionAppPath, ui);
-    let templateFilter: TemplateFilter = await getTemplateFilter(functionAppPath);
+    if (language === undefined) {
```

Comment:
!language
?

Don't you want the same functionality for null and ""?

Reply:
I rarely compare directly against undefined

Reply:
tslint doesn't like `!language` for enums because of the `!0` case (even though these are string enums, not number enums)

I'm fine with checking directly against undefined as long as strictNullChecks is turned on. You can't really make mistakes if the type checker complains about null vs undefined

Reply:
Sorry, missed that this was an enum, thought it was a string.